backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/comments/CommentEntryRepository.java

```diff
@@ -89,6 +108,17 @@ public void createCommentEntry(final @NonNull String characterData, @NonNull fin
 				.collect(GuavaCollectors.toImmutableList());
 	}
 
+	@NonNull
+	public Map<TableRecordReference, Boolean> hasComments(@NonNull final Collection<TableRecordReference> references)
+	{
+		final ImmutableMap.Builder<TableRecordReference, Boolean> result = ImmutableMap.builder();
+
+		referenceHasCommentsCache.getAllOrLoad(references, this::checkReferenceHasComments);
```

Comment:
pls use the return value of this function instead of stressing the cache for each value...

Reply:
`getAllOrLoad` returns a collection of booleans, and their order is not guaranteed. 
I can't really use the return value of the function call, because i cannot match the TRR with the respective result.

Reply:
clarified with teo. it's fine like this

Reply:
no no, pls:
* introduce `ReferenceSummary` private static class
* rename `referenceHasCommentsCache` to `referenceSummaryCache`

then u will be able to use it something like
```java
List<ReferenceComment> referenceSummaryList = referenceHasCommentsCache.getAllOrLoad(references, this::checkReferenceHasComments);

return referenceSummaryList.stream()
.collect(ImmutableMap.toImmutableMap(ReferenceComment::getRecordRef, ReferenceComment::hasComments));
```

Reply:
done